### ISSUE TITLE
Support struct variants in enums

### ISSUE THREAD

**Issue by Dgame** [PoiScript/strong-xml#3]
Currently it seems only tuple variants are supported:
```rust
#[derive(XmlWrite, XmlRead, PartialEq, Debug)]
#[xml(tag = "tag1")]
struct Tag1 { }

#[derive(XmlWrite, XmlRead, PartialEq, Debug)]
#[xml(tag = "tag2")]
struct Tag2 {}

#[derive(XmlWrite, XmlRead, PartialEq, Debug)]
enum Tag {
    #[xml(tag = "tag1")]
    Tag1(Tag1),
    #[xml(tag = "tag2")]
    Tag2(Tag2),
}
```

but I think struct variants would be really nice to have:
```rust
#[derive(XmlWrite, XmlRead, PartialEq, Debug)]
enum Tag {
    #[xml(tag = "tag1")]
    Tag1 { },
    #[xml(tag = "tag2")]
    Tag2 { }
}
```

**Comment by PoiScript** [comment]
This feature is definitely needed! But I don't have any information about how struct variant behaves in `syn`. So it probably takes more time to actually implement this.

**Comment by Dgame** [comment]
You could take a look at how serde does it. Or - to name an equivalent project - take a look at [yaserde](https://github.com/media-io/yaserde) especially at [the way they handle enums](https://github.com/media-io/yaserde/blob/6cf86c4e8ea9e8bb2faf35aa478330f721157e33/yaserde_derive/src/de/expand_enum.rs).

**Comment by PoiScript** [comment]
Thanks for suggestions! But unfortunately I don't have time for this right now, I'll come back to this on this weekend.

**Comment by PoiScript** [comment]
I just published v0.4.0, which supports struct variants, just give it a try!

**Comment by Dgame** [comment]
Great! Will do in the next few days. Thanks for your hard work :)
